crates/ln-dlc-node/src/fee_rate_estimator.rs

```diff
@@ -32,6 +33,43 @@ pub struct FeeRateEstimator {
     fee_rate_cache: RwLock<HashMap<ConfirmationTarget, FeeRate>>,
 }
 
+#[async_trait]
+pub trait EstimateFeeRate {
+    fn estimate(&self, target: ConfirmationTarget) -> FeeRate;
+    async fn update(&self) -> Result<()>;
```

Comment:
🤔 I don't think `update` needs to be part of this trait. The wallet only cares about getting the estimate, not how it is internally updated.